client/modules/IDE/components/CollectionList/CollectionList.jsx

```diff
@@ -1,313 +1,206 @@
-            actions={<SketchSearchbar />}
-            closeOverlay={this.hideAddSketches}
-            isFixedHeight
-          >
-            <AddToCollectionSketchList
-              collection={find(this.props.collections, {
-                id: this.state.addingSketchesToCollectionId
-              })}
-            />
-          </Overlay>
-        )}
-      </article>
-    );
-  }
-}
-
-CollectionList.propTypes = {
-  user: PropTypes.shape({
-    username: PropTypes.string,
-    authenticated: PropTypes.bool.isRequired
-  }).isRequired,
-  projectId: PropTypes.string,
-  getCollections: PropTypes.func.isRequired,
-  getProject: PropTypes.func.isRequired,
-  collections: PropTypes.arrayOf(
-    PropTypes.shape({
-      id: PropTypes.string.isRequired,
-      name: PropTypes.string.isRequired,
-      description: PropTypes.string,
-      createdAt: PropTypes.string.isRequired,
-      updatedAt: PropTypes.string.isRequired
-    })
-  ).isRequired,
-  username: PropTypes.string,
-  loading: PropTypes.bool.isRequired,
-  toggleDirectionForField: PropTypes.func.isRequired,
-  resetSorting: PropTypes.func.isRequired,
-  sorting: PropTypes.shape({
-    field: PropTypes.string.isRequired,
-    direction: PropTypes.string.isRequired
-  }).isRequired,
-  project: PropTypes.shape({
-    id: PropTypes.string,
-    owner: PropTypes.shape({
-      id: PropTypes.string
-    })
-  }),
-  t: PropTypes.func.isRequired,
-  mobile: PropTypes.bool
+  return (
+    <article className="sketches-table-container">
+      <Helmet>
+        <title>{getTitle()}</title>
+      </Helmet>
+
+      {renderLoader()}
+      {renderEmptyTable()}
+      {hasCollections() && (
+        <table
+          className="sketches-table"
+          summary={t('CollectionList.TableSummary')}
+        >
+          <thead>
+            <tr>
+              {renderFieldHeader('name', t('CollectionList.HeaderName'))}
+              {renderFieldHeader(
+                'createdAt',
+                t('CollectionList.HeaderCreatedAt', {
+                  context: props.mobile ? 'mobile' : ''
+                })
+              )}
+              {renderFieldHeader(
+                'updatedAt',
+                t('CollectionList.HeaderUpdatedAt', {
+                  context: props.mobile ? 'mobile' : ''
+                })
+              )}
+              {renderFieldHeader(
+                'numItems',
+                t('CollectionList.HeaderNumItems', {
+                  context: props.mobile ? 'mobile' : ''
+                })
+              )}
+              <th scope="col"></th>
+            </tr>
+          </thead>
+          <tbody>
+            {collections.map((collection) => (
+              <CollectionListRow
+                mobile={props.mobile}
+                collection={collection}
+                user={user}
+                username={props.username}
+                onAddSketches={() => showAddSketches(collection.id)}
+                t={t}
```

Comment:
Personally I would use another `useTranslation` for this component rather than passing down `t` from the parent.  Though I don't actually have any docs to back up if one way is better than the other.  I tried a quick search to find some authoritative reference but all I got was [one person on StackOverflow.](https://stackoverflow.com/questions/65422274/react-i18next-passing-t-function-to-all-children-or-usetranslation-in-every-chi)

Reply:
i have seen a lot of times `t` is being passed to child components throughout the codebase.

Reply:
Yeah there is a lot of that in the old/existing code.  Usually when I rewrite components I change it.  But like I said I'm not certain it's actually better, it might just be my personal preference.